### PR TITLE
feat(stubs): py-nanoid

### PR DESCRIPTION
**Pull request by oliverlambson**
Implements stubs proposed for py-nanoid here: https://github.com/puyuan/py-nanoid/pull/30

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Type aliases are fine, but you'll need to prefix them with a leading underscore (`_Algorithm` instead of `Algorithm`) if they don't exist at runtime

File: pyrightconfig.stricter.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
         "stubs/ldap3",
         "stubs/Markdown",
         "stubs/mysqlclient",
+        "stubs/nanoid",
```
Comment (AlexWaygood) [inline pyrightconfig.stricter.json:62]: This is only necessary if some things in the stub are still missing type annotations. Your stubs are already complete, so this is unnecessary :)

```suggestion
```

Comment (oliverlambson) [reply pyrightconfig.stricter.json]: Ah, thanks!

**Comment by AlexWaygood** [comment]
no need to worry about a clean git history in the PR, by the way -- we squash-merge everything at typeshed. We actually sort-of prefer contributors to avoid force-pushing, as GitHub's UI makes it annoying to see what changed between commits (though I know it's not impossible)

**Comment by oliverlambson** [comment]
> no need to worry about a clean git history in the PR, by the way -- we squash-merge everything at typeshed. We actually sort-of prefer contributors to avoid force-pushing, as GitHub's UI makes it annoying to see what changed between commits (though I know it's not impossible)

Ah, sorry—habit, won't do it again 👍

File: pyrightconfig.stricter.json (hunk 1 above)
Comment (oliverlambson) [reply pyrightconfig.stricter.json]: yeah that was just me being sloppy, nice test suite caught it though!

File: stubs/nanoid/nanoid/method.pyi [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from collections.abc import Callable
+from typing_extensions import TypeAlias
+
+_Algorithm: TypeAlias = Callable[[int], bytearray | list[int]]
```
Comment (AlexWaygood) [inline stubs/nanoid/nanoid/method.pyi:4]: Eyeballing the source code, could this be slightly simpler and less restrictive?

```suggestion
from collections.abc import Callable, Sequence
from typing_extensions import TypeAlias

_Algorithm: TypeAlias = Callable[[int], Sequence[int]]
```

**Review by AlexWaygood — COMMENTED** [review]
Looks great, thanks! Just one small comment:

**Review by AlexWaygood — APPROVED** [review]
LGTM, thanks!

File: stubs/nanoid/nanoid/method.pyi (hunk 2 above)
Comment (oliverlambson) [reply stubs/nanoid/nanoid/method.pyi]: nice, that's definitely better 👍

**Comment by oliverlambson** [comment]
> LGTM, thanks!

Thanks for the review, appreciate it

File: stubs/nanoid/nanoid/generate.pyi [hunk 3]
```diff
@@ -0,0 +1,3 @@
+def generate(
+    alphabet: str = "_-0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ", size: int = 21  # noqa: Y053
```
Comment (Avasam) [inline stubs/nanoid/nanoid/generate.pyi:2]: This (and in `stubs/nanoid/nanoid/non_secure_generate.pyi`) are the only time we noqa Y053 . Was there a specific reason we kept this string in particular ?

Comment (AlexWaygood) [reply stubs/nanoid/nanoid/generate.pyi:2]: I guess the chief point of Y053 is to catch very long strings that are dynamically generated by a function call at import time (so might be different on each machine) and/or are just too long to be useful. Here it's obviously not going to be different depending on which machine you're using. I guess whether it's useful or not to have this default value available in an IDE context is arguable

Comment (Avasam) [reply stubs/nanoid/nanoid/generate.pyi:2]: It's not a major issue or anything, just wanted to make sure it was purposeful and not glanced over during review !

**Pull request merged**